src/handlers/input_resources/ThreadMonitor.h

```diff
@@ -46,17 +48,20 @@ class ThreadMonitor
         }
         system_total_time = system_total_time / sysconf(_SC_NPROCESSORS_ONLN);
 
-        std::vector<uint64_t> stats;
+        std::vector<std::string> stats;
         std::ifstream thread_stat("/proc/thread-self/stat");
-        thread_stat.ignore(' ');
-        while (thread_stat >> stat) {
-            stats.push_back(stat);
+        std::string stat_str;
+        while (thread_stat >> stat_str) {
+            stats.push_back(stat_str);
+            if (stats.size() > PROC_STAT_POS_STIME)
```

Comment:
proc stat has more than 40 items and we only use the 13 and 14, so I think it is less **costly** to check size 15 times than pushing back string 25 times without being used